lib/modules/apostrophe-schemas/public/js/user.js

```diff
@@ -572,6 +572,46 @@ apos.define('apostrophe-schemas', {
     });
 
     self.addFieldType({
+      name: 'object',
+      populate: function populate(data, name, $field, $el, field, callback) {
+        var $fieldset = self.findFieldset($el, name);
+        _.each(field.schema, function (subField) {
+          var $input = self.findSafeInFieldset($fieldset, subField.name, '[name=' + subField.name + ']');
+          var $subFieldSet = self.findFieldset($fieldset, subField.name);
+
+          if (subField.required === true) {
+            $subFieldSet.addClass('apos-required');
+          }
+
+          if (subField.type === 'object') {
+            // recursive call : no need for callback as the real callback will be called at the end of this loop
+            populate(data[name], subField.name, $subFieldSet, $fieldset, subField, function() { return; })
+          } else {
+            $input.val(data[name][subField.name]);
+          }
+        })
+
+        return setImmediate(callback);
+      },
+      convert: function convert(data, name, $field, $el, field, callback) {
+        var $fieldset = self.findFieldset($el, name);
+        _.each(field.schema, function (subField) {
+          var $input = self.findSafeInFieldset($fieldset, subField.name, '[name=' + subField.name + ']');
+
+          if (subField.type === 'object') {
+            var $subFieldSet = self.findFieldset($fieldset, subField.name);
+            // recursive call : no need for callback as the real callback will be called at the end of this loop
+            convert(data[name], subField.name, $subFieldSet, $fieldset, subField, function() { return; })
```

Comment:
This code needs to invoke apos.schemas.convert() with the schema of the object field, so all field types work, all callbacks are invoked, proper sanitization occurs, etc.